writing DO csvs and making summary plots

jsadler2 opened the pull request:
- Writing out csvs for instantaneous and daily DO data. This is done for convenience so I can read the data into Python.
- Code that makes plots in #9

lekoenig commented on 1_fetch/src/write_data.R:
```diff
@@ -0,0 +1,7 @@
+library(readr)
```
I think the code should run without the explicit call to `readr` here since the tidyverse collection of packages is included in `tar_option_set(...)` within the `_targets.R` file.

jsadler2 replied:
Oh. Cool! I did not realize that.

lekoenig commented on 1_fetch/src/write_data.R:
```diff
@@ -0,0 +1,7 @@
+write_to_csv <- function(data, outfile){
+  write_csv(data, file = outfile)
+  return(outfile)
+}
```
I wondered whether this function is so small that it doesn't warrant a separate function call. However, I think the alternative would be to code the `write_csv` function within each target since `targets` is looking for the outfile path:

    tar_target(p1_daily_data_csv,  
    {write_to_csv <- function(data,outfile){  
    readr::write_csv(data,outfile)  
    return(outfile)}   
    write_to_csv(p1_daily_data,"1_fetch/out/daily_do_data.csv")  
    },  
    format = "file")

Since this would have to be repeated for p1_inst_data_csv, I think the separate `write_to_csv` file is more readable and can be repeated for the daily and inst datasets.

jsadler2 replied:
Yeah that was exactly the thought process I went through too. Glad to know I wasn't too off-base. I think it's just enough of a difference from the `readr` function that it would kind of clog up the pipeline file.

lekoenig commented on 3_visualize/src/do_overview_plots.py:
```diff
@@ -0,0 +1,134 @@
+    ax.legend(loc="upper left")
+
+fig = plt.gcf()
+fig.suptitle("daily minimum DO (mg/l) at daily sites", y=0.99)
```
Should this title read "daily mean DO (mg/l) at inst sites"?

jsadler2 replied:
oh yep! good catch.

jsadler2 commented:
Hi @lekoenig - thanks for the review. I think I addressed your comments.

I also wanted to point out for future reference that I had the idea to add the python plotting file as a dependency to the plotting target: https://github.com/USGS-R/drb-do-ml/pull/24/commits/f2f0ab7f8f0d213a85cf61cef7b1fa04fd0e1087. This way, targets knows to rebuild if I change the .py file (like if I mislabel the an axis label 😄).
